go-controller/pkg/node/openflow_manager.go

```diff
@@ -245,7 +245,38 @@ func (c *openflowManager) updateBridgeFlowCache(hostIPs []net.IP, hostSubnets []
 	}
 	dftFlows = append(dftFlows, dftCommonFlows...)
 
-	c.updateFlowCacheEntry("NORMAL", []string{fmt.Sprintf("table=0,priority=0,actions=%s\n", util.NormalAction)})
+	if config.OvnKubeNode.Mode == types.NodeModeDPU {
+		// Retrieve the local_ip for the tunnel endpoint which will be used to create OpenFlow actions
+		checkCmd := []string{
+			"get",
+			"Open_vSwitch",
+			".",
+			"external_ids:ovn-encap-ip",
+		}
+		encapIP, _, err := util.RunOVSVsctl(checkCmd...)
+		if err != nil {
+			return err
+		}
+		encapIP = strings.TrimSuffix(encapIP, "\n")
+
+		// The extra actions, compared to the full mode, are due to a DHCP server on the bridge.
+		// While not impacting the other flows negatively it implies something "custom"
+		// this could be a future feature to allow users to add custom flow actions on the managed ovnkube bridge via Kubernetes API.
+		c.updateFlowCacheEntry("NORMAL", []string{
+			// Add normal action on table 0 (act as a switch)
+			fmt.Sprintf("table=0,priority=0,actions=%s\n", util.NormalAction),
+			// Add OpenFlow action with high priority that allows a normal action for DHCP servers
+			// that are serving on the ovnkube managed bridge.
+			fmt.Sprintf("table=0,priority=999,udp,tp_dst=67,actions=%s\n", util.NormalAction),
+			// Add OpenFlow action with high priority that allows the PF (physical function) to the
+			// local_ip.
+			fmt.Sprintf("table=0,priority=998,ip,nw_dst=%s,actions=%s\n", encapIP, util.NormalAction),
```

Comment:
it's not clear to me who's IP the encap IP is. Is it the host PF? or the br-ovn DPU? The first flow says if it is a packet sent to a dhcp server then normal action. The second flow says if its a packet to the encap IP normal action. It's not making sense to me what the purpose of the 2nd flow is.

Maybe putting some IPs in your diagram and showing which one is which, and the analogy to br-ex in a regular deployment will help me understand.

Reply:
Note we could also update the same DPU markdown file with your diagram and explaining the DHCP server, what it is used for and where the IP addresses live for encap, what flows are added and why they are needed. Similar to:
https://github.com/ovn-kubernetes/ovn-kubernetes/blob/master/docs/design/service-traffic-policy.md#local-gateway-mode

Reply:
`it's not clear to me who's IP the encap IP is. Is it the host PF? or the br-ovn DPU? `
br-ovn
`The second flow says if its a packet to the encap IP normal action. It's not making sense to me what the purpose of the 2nd flow is.`
The issue is when I send traffic from the PF to the VTEP w/o this action it will go out the uplink instead of going to br-ovn

will add a picture to try to make it clearer.